pkg/network/setup/netstat.go

```diff
@@ -75,40 +75,45 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 		return nil
 	}
 
-	interfacesStatus, err := c.ifacesStatusFromPodCache(vmi)
-	if err != nil {
-		return err
-	}
-
-	if len(domain.Spec.Devices.Interfaces) > 0 {
-		interfacesStatus = ifacesStatusFromDomain(interfacesStatus, domain.Spec.Devices.Interfaces)
-	}
+	interfacesStatus := ifacesStatusFromDomain(domain.Spec.Devices.Interfaces)
 
+	var err error
 	if len(domain.Status.Interfaces) > 0 {
-		domain.Status.Interfaces = MergeAgentStatusesWithDomainData(domain.Spec.Devices.Interfaces, domain.Status.Interfaces)
+		domain.Status.Interfaces = mergeAgentStatusesWithDomainData(domain.Spec.Devices.Interfaces, domain.Status.Interfaces)
 		interfacesStatus = ifacesStatusFromGuestAgent(interfacesStatus, vmi.Spec.Domain.Devices.Interfaces, domain.Status.Interfaces)
+
+		ifacesSpec := netvmispec.FilterInterfacesSpec(vmi.Spec.Domain.Devices.Interfaces, func(i v1.Interface) bool {
+			return i.Masquerade != nil || i.Slirp != nil
+		})
+		if interfacesStatus, err = c.updateIfacesStatusFromPodCache(interfacesStatus, ifacesSpec, vmi); err != nil {
+			return err
+		}
+	} else {
+		interfacesStatus, err = c.updateIfacesStatusFromPodCache(interfacesStatus, vmi.Spec.Domain.Devices.Interfaces, vmi)
+		if err != nil {
+			return err
+		}
 	}
 
 	vmi.Status.Interfaces = interfacesStatus
 	return nil
 }
 
-func (c *NetStat) ifacesStatusFromPodCache(vmi *v1.VirtualMachineInstance) ([]v1.VirtualMachineInstanceNetworkInterface, error) {
-	ifacesStatus := []v1.VirtualMachineInstanceNetworkInterface{}
-	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.SRIOV != nil {
+// updateIfacesStatusFromPodCache updates the provided interfaces statuses with data (IP/s) from the pod-cache.
+func (c *NetStat) updateIfacesStatusFromPodCache(ifacesStatus []v1.VirtualMachineInstanceNetworkInterface, ifacesSpec []v1.Interface, vmi *v1.VirtualMachineInstance) ([]v1.VirtualMachineInstanceNetworkInterface, error) {
+	for _, iface := range ifacesSpec {
+		ifaceStatus := netvmispec.LookupInterfaceStatusByName(ifacesStatus, iface.Name)
+		if ifaceStatus == nil {
 			continue
 		}
+
 		podIface, err := c.getPodInterfacefromFileCache(vmi, iface.Name)
```

Comment:
If I am not mistaken, calling `netvmispec.LookupInterfaceStatusByName(..)` will return a `nil` for SR-IOV interfaces (as they will not appear in the status in case of no-ga).

But most importantly, we should have this covered in the unit tests. If we do not, we should fix it.